File: CS/WindowsFormsApplication1/CustomToggleSwitchPainter.cs
using DevExpress.Utils.Drawing;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;

namespace WindowsFormsApplication1
{
    class CustomToggleSwitchPainter : ToggleSwitchPainter
    {
        public CustomToggleSwitchPainter()
            : base()
        { }

        protected override void DrawContent(ControlGraphicsInfoArgs info)
        {
            BaseCheckEditViewInfo vi = info.ViewInfo as BaseCheckEditViewInfo;
            vi.CheckInfo.Appearance.TextOptions.HAlignment = DevExpress.Utils.HorzAlignment.Center;

            ToggleObjectInfoArgs args = (vi.CheckInfo) as ToggleObjectInfoArgs;

            if (args.GlyphRect.X == args.SwitchRect.X)
                vi.CheckInfo.CaptionRect = new Rectangle(args.SwitchRect.Right, args.SwitchRect.Y, args.SwitchRect.Width, args.SwitchRect.Height);
            else
                if (args.GlyphRect.Right == args.SwitchRect.Right)
                    vi.CheckInfo.CaptionRect = new Rectangle(args.GlyphRect.X, args.GlyphRect.Y, args.SwitchRect.Width, args.SwitchRect.Height);
                      
            base.DrawContent(info);
        }
    }
}

File: VB/WindowsFormsApplication1/CustomToggleSwitchPainter.vb
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo
Imports System.Drawing

Namespace WindowsFormsApplication1

    Friend Class CustomToggleSwitchPainter
        Inherits ToggleSwitchPainter

        Public Sub New()
            MyBase.New()
        End Sub

        Protected Overrides Sub DrawContent(ByVal info As ControlGraphicsInfoArgs)
            Dim vi As BaseCheckEditViewInfo = TryCast(info.ViewInfo, BaseCheckEditViewInfo)
            vi.CheckInfo.Appearance.TextOptions.HAlignment = DevExpress.Utils.HorzAlignment.Center
            Dim args As ToggleObjectInfoArgs = TryCast(vi.CheckInfo, ToggleObjectInfoArgs)
            If args.GlyphRect.X = args.SwitchRect.X Then
                vi.CheckInfo.CaptionRect = New Rectangle(args.SwitchRect.Right, args.SwitchRect.Y, args.SwitchRect.Width, args.SwitchRect.Height)
            ElseIf args.GlyphRect.Right = args.SwitchRect.Right Then
                vi.CheckInfo.CaptionRect = New Rectangle(args.GlyphRect.X, args.GlyphRect.Y, args.SwitchRect.Width, args.SwitchRect.Height)
            End If

            MyBase.DrawContent(info)
        End Sub
    End Class
End Namespace
